authapi/tests/test_organizations.py

```diff
@@ -767,8 +676,196 @@ def test_create_permission_for_organizations_team(self):
+                args=[org.pk, team.pk]
+            ),
+            data={
+                'type': 'organization:admin',
+                'object_id': org.pk,
+                'namespace': 'foo'
+            })
+        self.assertEqual(resp.status_code, status.HTTP_401_UNAUTHORIZED)
+
+    def test_permission_create_permission_no_permission(self):
+        '''Users that do not have the correct permissions should be denied.'''
+        user, token = self.create_user()
+        self.client.credentials(HTTP_AUTHORIZATION='Token ' + token.key)
+        org = SeedOrganization.objects.create(title='test org')
+        team = SeedTeam.objects.create(title='test team', organization=org)
+
+        resp = self.client.post(
+            reverse(
+                'seedorganization-teams-permissions-list',
+                args=[org.pk, team.pk]
+            ),
+            data={
+                'type': 'foo:bar',
+                'object_id': '7',
+                'namespace': 'foo'
+            })
+        self.assertEqual(resp.status_code, status.HTTP_403_FORBIDDEN)
+
+    def test_permission_create_permission_org_admin(self):
+        '''A user with org:admin should be able to create any permission for
+        that organization's teams.'''
+        user, token = self.create_user()
+        self.client.credentials(HTTP_AUTHORIZATION='Token ' + token.key)
+        org1 = SeedOrganization.objects.create(title='test org')
+        org2 = SeedOrganization.objects.create(title='test org')
+        team1 = SeedTeam.objects.create(title='test team', organization=org1)
+        team2 = SeedTeam.objects.create(title='test team', organization=org2)
+        self.add_permission(user, 'org:admin', org1.pk)
+
+        # Wrong org
+        resp = self.client.post(
+            reverse(
+                'seedorganization-teams-permissions-list',
+                args=[org2.pk, team1.pk]
+            ),
+            data={
+                'type': 'org:admin',
+                'object_id': org1.pk,
+                'namespace': 'foo'
+            })
+        self.assertEqual(resp.status_code, status.HTTP_404_NOT_FOUND)
+
+        # Wrong team
+        resp = self.client.post(
+            reverse(
+                'seedorganization-teams-permissions-list',
+                args=[org1.pk, team2.pk]
+            ),
+            data={
+                'type': 'org:admin',
+                'object_id': org1.pk,
+                'namespace': 'foo'
+            })
+        self.assertEqual(resp.status_code, status.HTTP_404_NOT_FOUND)
+
+        # Wrong org and team
+        resp = self.client.post(
+            reverse(
+                'seedorganization-teams-permissions-list',
+                args=[org2.pk, team2.pk]
+            ),
+            data={
+                'type': 'org:admin',
+                'object_id': org1.pk,
+                'namespace': 'foo'
+            })
+        self.assertEqual(resp.status_code, status.HTTP_403_FORBIDDEN)
```

Comment:
Hmm, if I'm understanding correctly, sounds like we need two separate test cases:
  - one for testing that we check the right team permissions when we create
  - one for testing that we check the right team permissions when we destroy

I think they make more sense outside of this test case since this is for testing adding of permissions for users with `org:admin`, while checking the right team permissions is something that happens regardless of the user's permissions.